dio/lib/src/dio.dart

```diff
@@ -43,18 +43,18 @@ import 'entry_stub.dart'
 ///  ```
 
 abstract class Dio {
-  factory Dio([BaseOptions options]) => createDio(options);
+  factory Dio([BaseOptions? options]) => createDio(options);
 
   /// Default Request config. More see [BaseOptions] .
-  BaseOptions options;
+  BaseOptions? options;
 
   Interceptors get interceptors;
 
-  HttpClientAdapter httpClientAdapter;
+  late HttpClientAdapter httpClientAdapter;
 
   /// [transformer] allows changes to the request/response data before it is sent/received to/from the server
   /// This is only applicable for request methods 'PUT', 'POST', and 'PATCH'.
-  Transformer transformer;
+  late Transformer transformer;
```

Comment:
Bonus points if you can make `options`, `httpClientAdapter` and `transformer` not be `late`; It's just because they are always initialized right away in constructors, which in most cases can be somehow rewritten to not require `late`. I don't think that change is absolutely necessary, though.